spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/RepositoryRestHandlerMapping.java

```diff
@@ -155,6 +191,36 @@ protected ProducesRequestCondition customize(ProducesRequestCondition condition)
 		return new ProducesRequestCondition(mediaTypes.toArray(new String[mediaTypes.size()]));
 	}
 
+	/* (non-Javadoc)
+	 * @see org.springframework.web.servlet.handler.AbstractHandlerMethodMapping#getCorsConfiguration(java.lang.Object, javax.servlet.http.HttpServletRequest)
+	 */
+	@Override
+	protected CorsConfiguration getCorsConfiguration(Object handler, HttpServletRequest request) {
+
+		CorsConfiguration corsConfiguration = super.getCorsConfiguration(handler, request);
+		String lookupPath = getUrlPathHelper().getLookupPathForRequest(request);
+
+		String repositoryLookupPath = new BaseUri(configuration.getBaseUri()).getRepositoryLookupPath(lookupPath);
+
+		// Repository root resource
+		if (StringUtils.hasText(repositoryLookupPath) && repositories != null) {
```

Comment:
Can we invert the conditions to be able to use early returns and thus avoid the nesting?